### DISCUSSION TITLE
Nuxt 3 extending the backend

### DISCUSSION THREAD

**Discussion by laflinwalker** [prisma/prisma#16135] [Q&A]
[nuxt] [request error] [unhandled] [500] @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
  at new PrismaClient (./node_modules/.prisma/client/index.js:3:11)  
  at fetchProducts (./.nuxt/dev/index.mjs:528:18)  
  at ./.nuxt/dev/index.mjs:539:16  
  at ./node_modules/h3/dist/index.mjs:585:14  
  at Object.handler (./node_modules/h3/dist/index.mjs:796:12)  
  at Object.handler (./node_modules/h3/dist/index.mjs:634:31)  
  at processTicksAndRejections (node:internal/process/task_queues:96:5)  
  at async Server.toNodeHandle (./node_modules/h3/dist/index.mjs:698:7)

**Comment by nurul3101** [comment]
Hello @laflinwalker 👋 
Just to confirm, have you invoked the `npx prisma generate` command before invoking your database queries?
If yes, can you please provide some more information on how you ran into this error?

**Reply by laflinwalker** [reply]
I did one project server with Prisma. (Backend)
Then  another nuxt 3 app (frontend) and extended the backend in nuxt config file. 
I get the error, trying to access the API routes

**Reply by nurul3101** [reply]
Can you provide a minimal reproduction so that I can investigate what could be the issue?

**Reply by laflinwalker** [reply]
//backend-product.js
 
import { PrismaClient } from '@prisma/client'

export const fetchProducts = async () => {
    const prisma = new PrismaClient()
    const products = await prisma.products.findMany({
        select: {
            name: true,
            productId: true,
        }
    })
    return products
}


//frontend-nuxt.config.ts

export default defineNuxtConfig({

    extends:['../backend']

})

//frontend-app.vue

<script setup>
const products = ref([]);
onMounted(async () => {
  products.value = await useFetch(() => "api/products");
});
</script>

<template>
  <div>
    {{ products }}
  </div>
</template>

**Reply by pantharshit00** [reply]
Maybe can you look into this example and match setup: https://github.com/ruheni/prisma-nuxt

I wasn't able to reproduce this.

**Reply by pantharshit00** [reply]
Hello @laflinwalker 

Is this still relevant with the latest version?

**Reply by pocho50** [reply]
Hello i m also using nuxt 3 but even if i invoke npx prisma generate when i deploy to firebase hosting i have an error @prisma/client did not initialize yet. In local is working perfect

**Reply by nurul3101** [reply]
@pocho50 Can you please open a new [GitHub Discussion](https://github.com/prisma/prisma/discussions/new/choose) with all relevant information so that we can debug?

**Comment by csjophon** [comment]
Hello, has this problem been solved? I called npx prisma generate successfully, but the nginx reverse proxy still failed to be deployed on the server.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
